modules/api/src/main/java/org/apache/ignite/sql/async/AsyncClosableCursor.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.sql.async;
+
+import java.util.concurrent.CompletableFuture;
+import org.apache.ignite.sql.NoRowSetExpectedException;
+
+/**
+ * Closeable cursor.
+ *
+ * @param <T> Type of elements.
+ */
+public interface AsyncClosableCursor<T> {
+    /**
+     * Returns the current page content if the query returns rows.
+     *
+     * @return Iterable set of rows.
+     * @throws NoRowSetExpectedException if no row set is returned.
+     */
+    Iterable<T> currentPage();
+
+    /**
+     * Returns the current page size if the query return rows.
+     *
+     * @return The size of {@link #currentPage()}.
+     * @throws NoRowSetExpectedException if no row set is returned.
+     */
+    int currentPageSize();
+
+    /**
+     * Fetches the next page of results asynchronously.
+     * The future that is completed with the same {@code AsyncResultSet} object.
```

Comment:
what does it mean 'completed with the same object'?

Reply:
This javadoc was moved from `AsyncResultSet#fetchNextPage()`. This means that future return `this`.